src/raspi/component/motor/odrive_motor.py

```diff
@@ -0,0 +1,136 @@
+import cantools
+import math
+import time
+import logging
+from dataclasses import dataclass
+from component.motor.odrive_enums import *
+
+import component.can.can_bus as can_bus
+
+from state_management import (
+    create_generic_context,
+    device,
+    device_action,
+    device_parser,
+    identifier
+)
+
+db = cantools.database.load_file("src/raspi/component/motor/odrive-cansimple.dbc")
+
+@device
+@dataclass()
+class ODriveMotor:
+    axisID: int
+    bus: can_bus.CanBus = identifier(can_bus.ctx)
+    current_state : MotorState = MotorState.UNDEFINED
+    current_position: float = None
+    current_velocity: float = None
+    default_state : MotorState = MotorState.CLOSED_LOOP_CONTROL
+
+    def __post_init__(self):
+        can_bus.add_listener(self.bus, self.axisID, lambda msg: read_message(self, msg))
+
+def read_message(motor, msg):
+    msg_id = msg.arbitration_id & 0x1F
+    msg_db = db.get_message_by_frame_id(msg_id)
+    match msg_db.name:
+        case "Heartbeat": #0x01
+            data = msg_db.decode(msg.data)
+            state = data['Axis_State'].value
+            motor.current_state = state
+            logging.info("Heartbeat w/ state: " + str(state))
+            axis_error = data['Axis_Error']
+            print
+            if get_error_num(axis_error) != 0:
+                logging.error("Axis Error w/ following data: " + axis_error)
+        case "Get_Motor_Error": #0x03
+            logging.error("Motor Error w/ following data: " + msg_db.decode(msg.data))
+        case "Get_Encoder_Error": #0x04
+            logging.error("Encoder Error w/ following data: " + msg_db.decode(msg.data))
+        case "Get_Sensorless_Error": #0x05
+            logging.error("Sensorless Error w/ following data: " + msg_db.decode(msg.data))
+        case "Get_Encoder_Estimates": #0x09
+            data = msg_db.decode(msg.data)
+            motor.current_position = data['Pos_Estimate']
+            motor.current_velocity = data['Vel_Estimate']
+            print("Estimates Updated")
+            pass
+        case "Get_Encoder_Count": #0x0A
+            pass
+        case "Get_IQ": #0x14
+            pass
+        case "Get_Sensorless_Estimates": #0x15
+            pass
+        case "Get_Bus_Voltage_Current": #0x17
+            pass
+        case "Get_ADC_Voltage": #0x1C
+            pass
+        case "Get_Controller_Error": #0x1D
+            logging.error("Controller Error w/ following data: " + msg_db.decode(msg.data))
+        case _:
+            logging.warning(f"Unrecognized message \"{msg_db.name}\" recieved (id {hex(msg_id)})")
```

Comment:
```suggestion
EVENT_HANDLE = dict()
EVENT_IGNORE = ["MotorStatus"]


def event_handle(motor, msg):
    msg_id = msg.arbitration_id & 0x1F
    msg_db = db.get_message_by_frame_id(msg_id)
    if msg_db.name in EVENT_HANDLE:
        EVENT_HANDLE[msg_db.name](motor, msg)
    elif msg_db.name not in EVENT_IGNORE:
        logging.warning(
            f'Unrecognized message "{msg_db.name}" recieved (id {hex(msg_id)})'
        )


def event_decorator(name):
    def wrapper(func):
        EVENT_HANDLE[name] = func
        return func

    return wrapper


@event_decorator("Heartbeat")
def motor_status(motor, msg):
    # Do something with the message
    pass


@event_decorator("Get_Motor_Status")
def motor_status(motor, msg):
    # Do something with the message
    pass

```
This is a more readable syntax for declaring this instead of a 40 line function with all of the event edgecases.